src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -798,12 +711,41 @@ public int getCurrentPoolSize() {
      * @return current number of workers executing tasks
      */
     public int getCurrentRunningCount() {
-      synchronized (workersLock) {
-        int workerOutQty = currentPoolSize - availableWorkers.size();
-        if (waitingForWorker || currentPoolSize == 0) {
-          return workerOutQty;
+      int startPoolSize = currentPoolSize.get();
```

Comment:
@lwahlmeier I think this is the last thing I need the thumbs up on before merging.  I removed the lock that we otherwise would have used here (and initial benchmarks look good, though they have not all ran yet).  So because of that we have to wait for a consistent state.

What this does is this:
* Total idle workers and track the chain state by doing ^ to the hash codes of each worker (defaults to Object.hashCode)
* We then loop over the idle workers again, doing the same operation
* If the pool size has not changed, and the recorded values between the loops are the same we return the result
* If the pool size has not changed, but the workers have either changed count or order, we then set the initial values to be our verification values, and re-run the verification (so in theory we will only need to loop once more in this case)
* If the pool size changed we must do the initial loop and verification loop again

This should give us a reliable number here, but of course at the cost of speed, particularly if there is a large quantity of idle threads.  I benchmarked it for 2000 idle threads with no retry contention and we are around .5 milliseconds.  At 5000 idle threads it takes around 1 millisecond to calculate.

Of course an AtomicInteger would cause this call to be super fast, but would end up causing a (granted small) impact at task execution and completion time.